### ISSUE TITLE
Wrong diagnostic about shadowing in nested class

### ISSUE THREAD

**Issue by gennaroprota** [llvm/llvm-project#62588]
Hi,

the following is a reduced test case of a situation I encountered in one of my libraries:

```c++
class container
{
public:
    char *              end();

    class nested
    {
    public:
        enum                end_tag { end };
    };    
};
```

Compiling this with Clang 16 and `-Wshadow` gives:

```console
<source>:9:39: warning: declaration shadows a static data member of 'container' [-Wshadow]
        enum                end_tag { end };
                                      ^
<source>:4:25: note: previous declaration is here
    char *              end();
                        ^
```
The warning seems totally off, because `container::end()` is neither static nor a data member.

Also, if `container::end()` is made static, the text of the warning still says "data member". (It is IMHO correct to warn, in that case, but the text of the warning should just say "member", not "data member".)

**Comment by shafik** [comment]
Confirmed, this looks like a quick fix.
